spec/models/asset_spec.rb

```diff
@@ -60,6 +60,19 @@
           expect(asset.errors[:replacement]).to include('not found')
         end
       end
+
+      context 'and the replacement exists but has been deleted' do
+        let(:replacement) { FactoryBot.create(:asset) }
+        let(:replacement_id) { replacement.id.to_s }
+
+        before do
+          replacement.destroy
```

Comment:
Minor: I think it would be better to use `replacement.destroy!` so we fail fast if the destroy fails for some unexpected reason.

Reply:
We can't do that as `#destroy!` actually destroys the object rather than just marking it as deleted.